test/functional/terminal/tui_spec.lua

```diff
@@ -151,7 +151,7 @@ describe('tui', function()
   it('can handle arbitrarily long bursts of input', function()
     execute('set ruler')
     local t = {}
-    for i = 1, 3000 do
+    for i = 2000, 3000 do
```

Comment:
Something to try: make the 3 feed() calls into one.

     feed('i\027[200~' .. table.concat(t, '\n') .. '\027[201~')

I have a feeling this is a quirk in the existing bracketed-paste logic.